src/main/java/picard/sam/AbstractAlignmentMerger.java

```diff
@@ -819,10 +819,10 @@ private static int getReadPositionAtReferencePositionIgnoreSoftClips(final SAMRe
 
         // Temporarily use the newCigar that has SOFT_CLIPs replaced with MATCH_OR_MISMATCH to get read position at reference, but ignore existence of soft-clips
         rec.setCigar(newCigar);
-        readPosition = SAMRecord.getReadPositionAtReferencePosition(rec, pos, false);
+        readPosition = SAMRecord.getReadPositionAtReferencePosition(rec, pos + posShift, false);
```

Comment:
add a comment that since the read effectively got shifted forward by turning the clips into matches, the query position needs also to be moved forward so that it's still querying the same base